### PR TITLE
chore: use github.com/Shopify/sarama Kafka models

### PR DESCRIPTION
**Pull request by smoya**
**Description**

https://github.com/Shopify/sarama is the most used Kafka client for Go. 
We were already using part of their model but copy pasted, since some interfaces needed for our project are private. 
We will also use Sarama for any kind of Kafka producing/consuming ops such as message routing, etc.

There has been some activity and predisposition to start working on the v2 of https://github.com/Shopify/sarama, which it might include some reworking around exposing Encoding/Decoding interfaces plus the possibility of other works that are also important to us (like exposing data  through interfaces, maybe something like the produce request records which now are private properties to become available somehow. This is totally needed for us).

That's why it made sense to me to create a fork of sarama in the meantime, with some of those changes, and reduce our code complexity and also reuse code. Once the development of v2 starts, I want to be involved and provide all our use cases and push for all those changes to happen somehow.

This PR uses that fork and removes all the copy/pasted code we had previously.

### CODE REVIEW

File: kafka/proxy.go [hunk 1]
```diff
@@ -131,19 +144,26 @@ func (h *produceRequestHandler) Handle(requestKeyVersion *kafkaprotocol.RequestK
...
+
 	for topic, r := range req.Records {
 		for _, s := range r {
 			if s.RecordBatch != nil {
 				for _, r := range s.RecordBatch.Records {
+					// Fixing indirection here
+					headers := make([]sarama.RecordHeader, len(r.Headers))
+					for i := 0; i < len(r.Headers); i++ {
+						headers[i] = *r.Headers[i]
+					}
```
Comment (magicmatatjahu) [inline kafka/proxy.go:158]: Why do you have to copy array if the type when copied matches?

**Review by magicmatatjahu — APPROVED** [review]

File: kafka/proxy.go (hunk 1 above)
Comment (smoya) [reply kafka/proxy.go:158]: They are different types. `[]sarama.RecordHeader` VS `[]*sarama.RecordHeader` 🤷

Comment (magicmatatjahu) [reply kafka/proxy.go:158]: Ahh ok, pointers :P

**Pull request merged**